File: api/ai/tags.py
from flask import Flask, request, jsonify
import os
import google.generativeai as genai

app = Flask(__name__)

# Configure Google AI
api_key = os.getenv('GOOGLE_API_KEY')
if api_key:
    genai.configure(api_key=api_key)
    model = genai.GenerativeModel('gemini-2.5-flash')
else:
    model = None

def handler(request):
    if request.method == 'OPTIONS':
        return ('', 200, {
            'Access-Control-Allow-Origin': '*',
            'Access-Control-Allow-Methods': 'POST, OPTIONS',
            'Access-Control-Allow-Headers': 'Content-Type'
        })
    
    if request.method != 'POST':
        return jsonify({'error': 'Method not allowed'}), 405
    
    try:
        data = request.get_json()
        if not data:
            return jsonify({'error': 'No content provided'}), 400
        
        text = data.get('text', '')
        if not text:
            return jsonify({'error': 'Text is required'}), 400
        
        if not model:
            return jsonify({'error': 'AI service not configured'}), 501
        
        max_tags = data.get('max', 10)
        prompt = f"Suggest up to {max_tags} relevant tags (keywords) for this text. Return only the tags separated by commas:\n\n{text}"
        response = model.generate_content(prompt)
        tags = [tag.strip() for tag in response.text.split(',') if tag.strip()]
        
        return jsonify({'tags': tags}), 200, {
            'Access-Control-Allow-Origin': '*'
        }
    except Exception as e:
        print(f"Error in tags endpoint: {e}")
        return jsonify({'error': 'Failed to suggest tags'}), 500, {
            'Access-Control-Allow-Origin': '*'
        }

File: api/ai/tags.js
const { GoogleGenerativeAI } = require('@google/generative-ai');

const GOOGLE_API_KEY = process.env.GOOGLE_API_KEY || 'AIzaSyCVnfhlcfFFVrRtA4tEHgCqACZX3-9QjjU';
let model = null;

try {
  if (GOOGLE_API_KEY) {
    const genAI = new GoogleGenerativeAI(GOOGLE_API_KEY);
    model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
  }
} catch (e) {
  model = null;
}

async function callGemini(prompt) {
  try {
    if (!model) {
      const text = prompt.split('\n\n').pop() || '';
      const words = text.toLowerCase().split(' ').filter(w => w.length > 3);
      const commonTags = ['notes', 'writing', 'important', 'personal', 'work', 'ideas', 'reminder', 'tasks'];
      const textTags = [...new Set(words.slice(0, 5))];
      return [...textTags, ...commonTags].slice(0, 8).join(', ');
    }
    const result = await model.generateContent(prompt);
    const response = await result.response;
    return response.text();
  } catch (error) {
    const text = prompt.split('\n\n').pop() || '';
    const words = text.toLowerCase().split(' ').filter(w => w.length > 3);
    const commonTags = ['notes', 'writing', 'important', 'personal', 'work', 'ideas', 'reminder', 'tasks'];
    const textTags = [...new Set(words.slice(0, 5))];
    return [...textTags, ...commonTags].slice(0, 8).join(', ');
  }
}

module.exports = async (req, res) => {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
  
  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }
  
  if (req.method !== 'POST') {
    return res.status(405).json({ error: 'Method not allowed' });
  }
  
  try {
    const { text, max = 10 } = req.body;
    const prompt = `Suggest up to ${max} relevant tags (keywords) for this text. Return only the tags separated by commas:\n\n${text}`;
    const response = await callGemini(prompt);
    const tags = response.split(',').map(tag => tag.trim()).filter(Boolean);
    res.json({ tags });
  } catch (error) {
    res.status(500).json({ error: 'Failed to generate tags' });
  }
};
